rpcdetect/jsonrpc.go

```diff
@@ -0,0 +1,138 @@
+package rpcdetect
+
+import (
+	"encoding/json"
+	"net/http"
+	"slices"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pokt-network/shannon-sdk/types"
+)
+
+var (
+	// defaultJSONRPCErrorReply is the default JSON-RPC error reply to be sent if the
+	// formatJSONRPCError function fails to format the appropriate one.
+	defaultJSONRPCErrorReply   *types.POKTHTTPResponse
+	defaultJSONRPCErrorReplyBz []byte
+)
+
+// init initializes the default JSON-RPC error reply. This function is called before
+// the main function and panics if it fails to marshal the default JSON-RPC error reply,
+// making the program exit early.
+func init() {
+	// Initialize the default JSON-RPC error reply
+	header := &types.Header{
+		Key:    contentTypeHeaderKey,
+		Values: []string{"application/json"},
+	}
+	headers := map[string]*types.Header{contentTypeHeaderKey: header}
+
+	defaultJSONRPCErrorReply = &types.POKTHTTPResponse{
+		StatusCode: http.StatusInternalServerError,
+		Header:     headers,
+		BodyBz:     []byte(`{"jsonrpc":"2.0","id":0,"error":{"code":-32000,"message":"Internal error","data":null}}`),
+	}
+
+	var err error
+	defaultJSONRPCErrorReplyBz, err = proto.Marshal(defaultJSONRPCErrorReply)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// jsonRPCPayload represents the JSON-RPC payload fields that are relevant for
```

Comment:
Are we just using this for detection or more?

1. If it's purely for detection -> ename to `jsonRPCPayloadMeta`
2. If it's for more -> #PUC

Ditto elsewhere.